### PR TITLE
refactor(infrastructure): Cut recompile time in half (12s to 5s)

### PR DESCRIPTION
**Pull request by acdvorak**
Don't build dist files in demo mode (they're not used).

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -170,58 +119,113 @@ module.exports = [{
...
+      toolbar: [path.resolve('./packages/mdc-toolbar/index.js')],
+    },
+    output: {
+      path: OUT_PATH,
+      publicPath: PUBLIC_PATH,
+      filename: 'mdc.[name].' + (IS_PROD ? 'min.' : '') + 'js',
+      libraryTarget: 'umd',
+      library: ['mdc', '[name]'],
+    },
+    // See https://github.com/webpack/webpack-dev-server/issues/882
```
Comment (kfranqueiro) [inline webpack.config.js]: Is the `devServer` block here and for `css` below even necessary anymore if we're not going to run these targets when we run the dev server?

Comment (kfranqueiro) [reply webpack.config.js]: (Additional thought, if we remove `devServer` on these, maybe move the comment that's above one of them to the first `devServer` config that's still relevant.)

Comment (acdvorak) [reply webpack.config.js]: Good call! Turns out you only need one `devServer` block, and it _must_ be in the first entry of the webpack config. It is completely ignored in later entries. Removed redundant blocks.

Comment (kfranqueiro) [reply webpack.config.js]: Seriously? Wouldn't have guessed that one... unintentionally dodged a bullet.

**Pull request merged**